### ISSUE TITLE
Refactor metaToJsonProperty to accept AJV keywords

### ISSUE THREAD

**Issue by iqbaldjulfri** [strongloop/loopback-next#2684]
## Description
We need to add additional attributes ([AJV keywords](https://ajv.js.org/keywords.html)) from `PropertyDefinition` so they will be written to `openapi.json` and we can use it as validation. Currently, only `type`, `description`, `items` (for arrays), and `$ref` are written (and a hard-coded `format: 'date-time'` for `Date` instances).

## Current Behavior
```ts
@model()
export class Employee extends Entity {
  @property({
    type: 'string',
    required: true,
    format: 'email', // this won't be written to openapi.json
  })
  email: string;

  @property({
    type: 'string',
    required: true,
    pattern: '(staff|manager|director)', // this also won't be written to openapi spec
  })
  type: string;
}
```

## Expected Behavior
Other AJV keywords is written to openapi spec

**Comment by iqbaldjulfri** [comment]
#2685 this is the pull request for this issue

**Comment by dhmlau** [comment]
@iqbaldjulfri, let me assign this issue to you since you have a PR up.

**Comment by bajtos** [comment]
I feel this is a duplicate of https://github.com/strongloop/loopback-next/issues/1624

**Comment by iqbaldjulfri** [comment]
the solution has been merged in #2685 .
